plugins/woocommerce-blocks/assets/js/blocks/product-filters/inner-blocks/price-filter/frontend.ts

```diff
@@ -2,232 +2,102 @@
-	const { decimalSeparator } = getCurrency();
-	const { minRange, minPrice, maxPrice, maxRange } = context;
-	const type = event.target.name;
-	const value = parseInt(
-		event.target.value
-			.replace( new RegExp( `[^0-9\\${ decimalSeparator }]+`, 'g' ), '' )
-			.replace( new RegExp( `\\${ decimalSeparator }`, 'g' ), '.' ),
-		10
-	);
-
-	const currentMinPrice =
-		type === 'min'
-			? Math.min( Number.isNaN( value ) ? minRange : value, maxPrice )
-			: minPrice;
-
-	const currentMaxPrice =
-		type === 'max'
-			? Math.max( Number.isNaN( value ) ? maxRange : value, minPrice )
-			: maxPrice;
-
-	if ( type === 'min' ) {
-		element.ref.value = currentMinPrice;
-	} else if ( type === 'max' ) {
-		element.ref.value = currentMaxPrice;
-	}
-
-	context.minPrice = currentMinPrice;
-	context.maxPrice = currentMaxPrice;
-
-	navigate(
-		getUrl( {
-			minRange,
-			maxRange,
-			minPrice: currentMinPrice,
-			maxPrice: currentMaxPrice,
-		} )
-	);
-}, 1000 );
-
-const constrainRangeSliderValues = (
-	/**
-	 * Tuple containing min and max values.
-	 */
-	values: [ number, number ],
-	/**
-	 * Min allowed value for the sliders.
-	 */
-	min?: number | null,
-	/**
-	 * Max allowed value for the sliders.
-	 */
-	max?: number | null,
-	/**
-	 * Step value for the sliders.
-	 */
-	step = 1,
-	/**
-	 * Whether we're currently interacting with the min range slider or not, so we update the correct values.
-	 */
-	isMin = false
-): { minPrice: number; maxPrice: number } => {
-	let [ minValue, maxValue ] = values;
-
-	const isFinite = ( n: number | null | undefined ): n is number =>
-		Number.isFinite( n );
-
-	if ( ! isFinite( minValue ) ) {
-		minValue = min || 0;
-	}
-
-	if ( ! isFinite( maxValue ) ) {
-		maxValue = max || step;
-	}
-
-	if ( isFinite( min ) && min > minValue ) {
-		minValue = min;
-	}
-
-	if ( isFinite( max ) && max <= minValue ) {
-		minValue = max - step;
-	}
-
-	if ( isFinite( min ) && min >= maxValue ) {
-		maxValue = min + step;
-	}
-
-	if ( isFinite( max ) && max < maxValue ) {
-		maxValue = max;
-	}
-
-	if ( ! isMin && minValue >= maxValue ) {
-		minValue = maxValue - step;
-	}
-
-	if ( isMin && maxValue <= minValue ) {
-		maxValue = minValue + step;
-	}
-
-	return { minPrice: minValue, maxPrice: maxValue };
-};
+const debounceUpdate = debounce(
+	(
+		context: PriceFilterContext,
+		parentContext: ProductFiltersContext,
+		price: Record< string, string >
+	) => {
+		Object.keys( price ).forEach( ( key ) => {
+			if ( key !== 'min_price' && key !== 'max_price' ) {
+				delete price[ key ];
+				return;
+			}
+			if ( ! price[ key ] ) {
+				delete price[ key ];
+			}
+		} );
 
-const getRangeStyle = (
-	minPrice: number,
-	maxPrice: number,
-	minRange: number,
-	maxRange: number
-) => {
-	return `--low: ${
-		( 100 * ( minPrice - minRange ) ) / ( maxRange - minRange )
-	}%; --high: ${
-		( 100 * ( maxPrice - minRange ) ) / ( maxRange - minRange )
-	}%;`;
-};
+		const params = parentContext.params;
```

Comment:
Considering that we are deleting the keys from the `parentContext.params` it would be better to copy everything to a new object, so we avoid mutating the original object.

```suggestion
		const params = {...parentContext.params };
```

Reply:
Good catch, I will update my code following this.